CMakeLists.txt

```diff
@@ -1,24 +1,28 @@
 
 cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 
-find_package(IRODS 4.2.11 EXACT REQUIRED)
+#find_package(IRODS 4.2.11 EXACT REQUIRED)
+find_package(IRODS REQUIRED)
 
-set(IRODS_PLUGIN_REVISION "0")
+set(IRODS_PLUGIN_REVISION "1")
 set(IRODS_PLUGIN_VERSION "${IRODS_VERSION}.${IRODS_PLUGIN_REVISION}")
 
+include(IrodsCXXCompiler)
+
 set(CMAKE_C_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang)
 set(CMAKE_CXX_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang++)
 set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
 
-project(irods_netcdf_plugin 
-VERSION "${IRODS_PLUGIN_VERSION}"
-LANGUAGES C CXX)
+#project(irods_netcdf_plugin VERSION "${IRODS_PLUGIN_VERSION}" LANGUAGES C CXX)
+project(irods_netcdf_plugin VERSION "4.3.1.1" LANGUAGES C CXX)
+
+include_directories("/usr/include/irods")
 
-set(CMAKE_CXX_FLAGS "-std=c++17 -stdlib=libc++ -Wno-write-strings ${CMAKE_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS "-std=c++20 -stdlib=libc++ -Wno-write-strings ${CMAKE_CXX_FLAGS}")
 
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-set(CMAKE_INSTALL_RPATH "/opt/irods-externals/clang-runtime6.0-0/lib:/opt/irods-externals/boost1.67.0-0/lib")
+set(CMAKE_INSTALL_RPATH "/opt/irods-externals/clang-runtime13.0-0/lib:/opt/irods-externals/boost1.77.0-1/lib")
```

Comment:
Wrong boost version.

Use boost1.81.0-0.

Reply:
Hm, wonder if that could be at the heart of the segment violation?

Reply:
Found out what the SIGSEGV crash was: `clearOutStruct` needs population (it's new in apidef_t in 4.3.1).